plugins/woocommerce/includes/admin/class-wc-admin-status.php

```diff
@@ -328,15 +354,29 @@ private static function flush_db_logs() {
 	 * @since 3.0.0
 	 */
 	private static function log_table_bulk_actions() {
-		if ( empty( $_REQUEST['_wpnonce'] ) || ! wp_verify_nonce( $_REQUEST['_wpnonce'], 'woocommerce-status-logs' ) ) { // WPCS: input var ok, sanitization ok.
-			wp_die( esc_html__( 'Action failed. Please refresh the page and retry.', 'woocommerce' ) );
+		check_admin_referer( 'bulk-logs' );
+
+		if ( ! current_user_can( 'manage_woocommerce' ) ) {
+			wp_die( esc_html__( 'You do not have permission to manage log entries.', 'woocommerce' ) );
 		}
 
-		$log_ids = array_map( 'absint', (array) isset( $_REQUEST['log'] ) ? wp_unslash( $_REQUEST['log'] ) : array() ); // WPCS: input var ok, sanitization ok.
+		$log_ids = filter_input(
+			INPUT_GET,
+			'log',
+			FILTER_CALLBACK,
+			array(
+				'options' => fn( $log ) => absint( $log ),
```

Comment:
If we do keep the `filter_input` I think this could just be `'options' => 'absint'`

Reply:
See 26c3ec844c5e3d72847a3c06aa6dfb53efd0f895 for a change to this.